### ISSUE TITLE
docs: enable link checking

### ISSUE THREAD

**Issue by pohly** [intel/pmem-csi#993]
We've got too many internal links that get broken when editing documentation simply because it is error-prone to find all references to anchors and/or files.

External links obviously can get broken by changes in the referenced sites.

We should enable a link checker to detect such problems.

**Comment by pohly** [comment]
Enabling it is not hard:
```
diff --git a/Makefile b/Makefile
index c33a5588d..cb93f1acb 100644
--- a/Makefile
+++ b/Makefile
@@ -244,7 +244,7 @@ $(addprefix test-kustomize-,$(KUSTOMIZE_OUTPUT)): test-kustomize-%: _work/kustom
 check-go-version-%:
        @ hack/verify-go-version.sh "$*"
 
-SPHINXOPTS    = -W --keep-going # Warn about everything, abort with an error at the end.
+SPHINXOPTS    = -b linkcheck -n -W --keep-going # Check links, be nitpicky. Warn about everything, abort with an error at the end.
 SPHINXBUILD   = sphinx-build
 SOURCEDIR     = .
 BUILDDIR      = _output
```

But it has false positives:
```
(docs/DEVELOPMENT: line   51) broken    /Makefile - 
```

A relative link (`../Makefile`) triggers a different error (`DEVELOPMENT.md:51: WARNING: None:any reference target not found: ../Makefile`) but the generated HTML and GitHub seem to be okay. It also works better in Emacs (can follow the link).

```
(docs/DEVELOPMENT: line   41) broken    autotest.md#configuration-options - 
```

I've not found a solution for this. This is a showstopper at the moment.

**Comment by pohly** [comment]
@intelkevinputnam can you perhaps help here?

**Comment by pohly** [comment]
I've started by forking the linkcheck.py.

**Comment by intelkevinputnam** [comment]
@pohly Sorry for the slow response. I'm headed on vacation at the end of the week, so I've been trying to get ahead of work already in progress. It's going to be challenging to get link checking to work because of the changes that sphinx-md makes to some URLs. I can't look at it right now, but I will be back in the office on July 1.

**Comment by pohly** [comment]
@intelkevinputnam: Don't worry, I think I can solve this myself.

**Comment by pohly** [comment]
The last commit in https://github.com/intel/pmem-csi/pull/1003/commits has my modifications to the original linkcheck.py.

I think it now works as intended. It reports several broken links and anchors and all of those indeed seem to be broken.

**Comment by pohly** [comment]
Solved with PR #1003.
